failing LL download and complete cleaning of install folder

**Describe the bug**

Two issues that seem to be related in some way:

1. This Loverslab file that is added as attachment in a comment will always fail, afaik for everyone:
SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip
https://www.loverslab.com/topic/103538-slal-animation-pack-by-leito-se/page/4/#comment-4286374
2. The install folder can get cleaned from all mods, even though it contains files that should be correct.

**Additional Context**
Steps to reproduce:
1. I made a TinyTestList compile of the above LL mod, USSEP and Mod organizer:
https://mega.nz/file/ToxWFAaa#3UpG0aZl6h3mCwS6IFeLlVEulgJBhTpKWpLM6xS3Yv4
2. Wabbajack will succeed downloading USSEP and Mod organizer, but fail the LL file SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip
[Wabbajack.00.log](https://github.com/user-attachments/files/19404180/Wabbajack.00.log)
```
00:00:05.103 [INFO] (Wabbajack.Downloaders.DownloadDispatcher) Finished downloading SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip. Hash: GzlTAsTd2RI=; Size: 7,8KB/22,6MB
00:00:05.103 [WARN] (Wabbajack.Downloaders.DownloadDispatcher) Initial download of SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip failed, trying mirror
00:00:05.103 [INFO] (Wabbajack.Downloaders.DownloadDispatcher) Downloading SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip from mirror, hash UA5KJMNCH7o=
00:00:05.103 [INFO] (Wabbajack.Downloaders.WabbajackCDNDownloader) Getting file definition for CDN download WabbajackCDNDownloader+State|https://mirror.wabbajack.org/500e4a24c3421fba
00:00:05.236 [INFO] (Wabbajack.Networking.NexusApi.ProxiedNexusApi) Nexus API call finished: 0 - Remaining Limit: 0
00:00:24.806 [INFO] (Wabbajack.Downloaders.DownloadDispatcher) Finished downloading SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip. Hash: AAAAAAAAAAA=; Size: 7,8KB/22,6MB
00:00:24.806 [ERROR] (Wabbajack.Installer.StandardInstaller) Downloaded hash AAAAAAAAAAA= does not match expected hash: UA5KJMNCH7o=
```
3. As workaround for the failing download. Download manually (https://www.loverslab.com/topic/103538-slal-animation-pack-by-leito-se/page/4/#comment-4286374), place it in the download folder. And restart wabbajack.
Wabbajack finishes the installation without problem.
[Wabbajack.01.log](https://github.com/user-attachments/files/19404182/Wabbajack.01.log)
`00:00:20.825 [INFO] (Wabbajack.Installer.StandardInstaller) Finished Installation`
4. Now delete the LL mod:
```
TinyTestList/mods/SLAL Animations by Leito v1.6
TinyTestList/downloads/SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip
TinyTestList/downloads/SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip.meta
```
5. Rerun wabbajack once more. It will fail on the LL again (as expected)
6. Don't close wabbajack, click on "Retry".
After retry, I see the files being deleted at the `Getting file definition for CDN download WabbajackCDNDownloader+State|https://mirror.wabbajack.org/500e4a24c3421fba` message. Video:
https://github.com/user-attachments/assets/871efb15-b678-409b-8e6e-9e493d3e11d0
[Wabbajack.02.log](https://github.com/user-attachments/files/19404245/Wabbajack.02.log)
```
00:00:03.776 [INFO] (Wabbajack.Installer.StandardInstaller) Install Folder: D:\Modlists\TinyTestList
00:00:03.776 [INFO] (Wabbajack.Installer.StandardInstaller) Downloads Folder: D:\Modlists\TinyTestList\downloads
00:00:03.776 [INFO] (Wabbajack.Installer.StandardInstaller) Game Folder: D:\Games\SteamLibrary\steamapps\common\Skyrim Special Edition
00:00:03.776 [INFO] (Wabbajack.Installer.StandardInstaller) Wabbajack Folder: D:\Wabbajack\4.0.0.3
00:00:03.780 [INFO] (Wabbajack.Installer.StandardInstaller) Optimizing ModList directives
00:00:03.803 [INFO] (Wabbajack.Installer.StandardInstaller) Next Step: Looking for files to delete
00:00:03.823 [INFO] (Wabbajack.Installer.StandardInstaller) Next Step: Cleaning empty folders
00:00:03.850 [INFO] (Wabbajack.Installer.StandardInstaller) Next Step: Looking for unmodified files
00:00:05.106 [INFO] (Wabbajack.Installer.StandardInstaller) Next Step: Updating ModList
00:00:05.106 [INFO] (Wabbajack.Installer.StandardInstaller) Optimized 2032 directives to 385 required
00:00:05.113 [INFO] (Wabbajack.Installer.StandardInstaller) Next Step: Hashing Archives
00:00:05.113 [INFO] (Wabbajack.Installer.StandardInstaller) Looking for files to hash
00:00:05.113 [INFO] (Wabbajack.Installer.StandardInstaller) Getting archive sizes
00:00:05.113 [INFO] (Wabbajack.Installer.StandardInstaller) Linking archives to downloads
00:00:05.124 [INFO] (Wabbajack.Installer.StandardInstaller) Found 107 total files, 0 matching filesize
00:00:05.124 [INFO] (Wabbajack.Installer.StandardInstaller) Missing 1 archives
00:00:05.124 [INFO] (Wabbajack.Installer.StandardInstaller) Downloading validation data
00:00:05.307 [INFO] (Wabbajack.Installer.StandardInstaller) Validating Archives
00:00:05.307 [INFO] (Wabbajack.Installer.StandardInstaller) Downloading missing archives
00:00:05.307 [INFO] (Wabbajack.Installer.StandardInstaller) Downloading 1 archives
00:00:05.307 [INFO] (Wabbajack.Installer.StandardInstaller) Next Step: Downloading files
00:00:05.315 [INFO] (Wabbajack.Installer.StandardInstaller) Downloading SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip
00:00:05.315 [DEBUG] (Wabbajack.Networking.Http.ResumableDownloader) Download for 'SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip' is starting from 0...
00:00:05.506 [INFO] (Wabbajack.Downloaders.DownloadDispatcher) Finished downloading SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip. Hash: JXZ+ruCUzjo=; Size: 7,8KB/22,6MB
00:00:05.506 [WARN] (Wabbajack.Downloaders.DownloadDispatcher) Initial download of SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip failed, trying mirror
00:00:05.506 [INFO] (Wabbajack.Downloaders.DownloadDispatcher) Downloading SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip from mirror, hash UA5KJMNCH7o=
00:00:05.506 [INFO] (Wabbajack.Downloaders.WabbajackCDNDownloader) Getting file definition for CDN download WabbajackCDNDownloader+State|https://mirror.wabbajack.org/500e4a24c3421fba
00:00:25.328 [INFO] (Wabbajack.Downloaders.DownloadDispatcher) Finished downloading SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip. Hash: AAAAAAAAAAA=; Size: 7,8KB/22,6MB
00:00:25.328 [ERROR] (Wabbajack.Installer.StandardInstaller) Downloaded hash AAAAAAAAAAA= does not match expected hash: UA5KJMNCH7o=
00:00:25.328 [INFO] (Wabbajack.Installer.StandardInstaller) Next Step: Hashing Archives
00:00:25.328 [INFO] (Wabbajack.Installer.StandardInstaller) Looking for files to hash
00:00:25.328 [INFO] (Wabbajack.Installer.StandardInstaller) Getting archive sizes
00:00:25.328 [INFO] (Wabbajack.Installer.StandardInstaller) Linking archives to downloads
00:00:25.328 [INFO] (Wabbajack.Installer.StandardInstaller) Found 107 total files, 0 matching filesize
00:00:25.328 [FATAL] (Wabbajack.Installer.StandardInstaller) Unable to download SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip (HTTPDownloader+State|https://www.loverslab.com/applications/core/interface/file/attachment.php?id=1924278&key=46468155102d24c004a61e4194e13080)
00:00:25.328 [FATAL] (Wabbajack.Installer.StandardInstaller) Cannot continue, was unable to download one or more archives
00:00:29.079 [INFO] (Wabbajack.InstallationVM) Preparing HttpDownloader
00:00:29.176 [INFO] (Wabbajack.InstallationVM) Starting installation of TinyTestList 1.0:
00:00:29.176 [INFO] (Wabbajack.InstallationVM)     Installation folder: D:\Modlists\TinyTestList
00:00:29.176 [INFO] (Wabbajack.InstallationVM)     Downloads folder: D:\Modlists\TinyTestList\downloads
00:00:29.176 [INFO] (Wabbajack.InstallationVM)     Modlist file location: D:\Modlists\Default.wabbajack
00:00:29.176 [INFO] (Wabbajack.InstallationVM)     Game: SkyrimSpecialEdition
00:00:29.176 [INFO] (Wabbajack.InstallationVM)     Game folder: D:\Games\SteamLibrary\steamapps\common\Skyrim Special Edition
00:00:29.176 [INFO] (Wabbajack.Installer.StandardInstaller) Installing: TinyTestList - 1.0
00:00:29.215 [INFO] (Wabbajack.Installer.StandardInstaller) Next Step: Configuring Installer
00:00:29.215 [INFO] (Wabbajack.Installer.StandardInstaller) Configuring Processor
00:00:29.215 [INFO] (Wabbajack.Installer.StandardInstaller) Install Folder: D:\Modlists\TinyTestList
00:00:29.215 [INFO] (Wabbajack.Installer.StandardInstaller) Downloads Folder: D:\Modlists\TinyTestList\downloads
00:00:29.215 [INFO] (Wabbajack.Installer.StandardInstaller) Game Folder: D:\Games\SteamLibrary\steamapps\common\Skyrim Special Edition
00:00:29.215 [INFO] (Wabbajack.Installer.StandardInstaller) Wabbajack Folder: D:\Wabbajack\4.0.0.3
00:00:29.215 [INFO] (Wabbajack.Installer.StandardInstaller) Optimizing ModList directives
00:00:29.221 [INFO] (Wabbajack.Installer.StandardInstaller) Next Step: Looking for files to delete
00:00:29.388 [INFO] (Wabbajack.Installer.StandardInstaller) Next Step: Cleaning empty folders
00:00:29.443 [INFO] (Wabbajack.Installer.StandardInstaller) Next Step: Looking for unmodified files
00:00:29.454 [INFO] (Wabbajack.Installer.StandardInstaller) Next Step: Updating ModList
00:00:29.454 [INFO] (Wabbajack.Installer.StandardInstaller) Optimized 385 directives to 385 required
00:00:29.454 [INFO] (Wabbajack.Installer.StandardInstaller) Next Step: Hashing Archives
00:00:29.454 [INFO] (Wabbajack.Installer.StandardInstaller) Looking for files to hash
00:00:29.454 [INFO] (Wabbajack.Installer.StandardInstaller) Getting archive sizes
00:00:29.454 [INFO] (Wabbajack.Installer.StandardInstaller) Linking archives to downloads
00:00:29.454 [INFO] (Wabbajack.Installer.StandardInstaller) Found 107 total files, 0 matching filesize
00:00:29.454 [INFO] (Wabbajack.Installer.StandardInstaller) Missing 1 archives
00:00:29.454 [INFO] (Wabbajack.Installer.StandardInstaller) Downloading validation data
00:00:29.570 [INFO] (Wabbajack.Installer.StandardInstaller) Validating Archives
00:00:29.570 [INFO] (Wabbajack.Installer.StandardInstaller) Downloading missing archives
00:00:29.570 [INFO] (Wabbajack.Installer.StandardInstaller) Downloading 1 archives
00:00:29.570 [INFO] (Wabbajack.Installer.StandardInstaller) Next Step: Downloading files
00:00:29.570 [INFO] (Wabbajack.Installer.StandardInstaller) Downloading SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip
00:00:29.570 [DEBUG] (Wabbajack.Networking.Http.ResumableDownloader) Download for 'SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip' is starting from 0...
00:00:30.259 [INFO] (Wabbajack.Downloaders.DownloadDispatcher) Finished downloading SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip. Hash: dmoC365G9zg=; Size: 125,1KB/22,6MB
00:00:30.259 [WARN] (Wabbajack.Downloaders.DownloadDispatcher) Initial download of SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip failed, trying mirror
00:00:30.259 [INFO] (Wabbajack.Downloaders.DownloadDispatcher) Downloading SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip from mirror, hash UA5KJMNCH7o=
00:00:30.259 [INFO] (Wabbajack.Downloaders.WabbajackCDNDownloader) Getting file definition for CDN download WabbajackCDNDownloader+State|https://mirror.wabbajack.org/500e4a24c3421fba
00:00:49.905 [INFO] (Wabbajack.Downloaders.DownloadDispatcher) Finished downloading SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip. Hash: AAAAAAAAAAA=; Size: 125,1KB/22,6MB
00:00:49.906 [ERROR] (Wabbajack.Installer.StandardInstaller) Downloaded hash AAAAAAAAAAA= does not match expected hash: UA5KJMNCH7o=
00:00:49.906 [INFO] (Wabbajack.Installer.StandardInstaller) Next Step: Hashing Archives
00:00:49.906 [INFO] (Wabbajack.Installer.StandardInstaller) Looking for files to hash
00:00:49.906 [INFO] (Wabbajack.Installer.StandardInstaller) Getting archive sizes
00:00:49.906 [INFO] (Wabbajack.Installer.StandardInstaller) Linking archives to downloads
00:00:49.906 [INFO] (Wabbajack.Installer.StandardInstaller) Found 107 total files, 0 matching filesize
00:00:49.906 [FATAL] (Wabbajack.Installer.StandardInstaller) Unable to download SLAL_AnimationsByLeito_v1.6_Fixed_SSE.zip (HTTPDownloader+State|https://www.loverslab.com/applications/core/interface/file/attachment.php?id=1924278&key=46468155102d24c004a61e4194e13080)
00:00:49.906 [FATAL] (Wabbajack.Installer.StandardInstaller) Cannot continue, was unable to download one or more archives

```

I don't think I clicked retry the first 2 times it deleted the files for me. But these steps reproduce the issue every time for me.

**Wabbajack Version**

4.0.0.3

**Logs**

--- a/Wabbajack.App.Wpf/ViewModels/Installers/InstallationVM.cs
+++ b/Wabbajack.App.Wpf/ViewModels/Installers/InstallationVM.cs
@@ -619,25 +619,31 @@ public class InstallationVM : ProgressViewModel, ICpuStatusVM
 
             try
             {
-                var canSource = GameRegistry.Games[ModList.GameType].CanSourceFrom ?? Array.Empty<Game>();
-                var namedgames = ModList.OtherGames;
+                // Always reload the modlist, incase of retrying , so it gets fresh directives and archives
+                var freshModList = await StandardInstaller.LoadFromFile(
+                    _dtos, WabbajackFileLocation.TargetPath);
+
+                var canSource = GameRegistry.Games[freshModList.GameType].CanSourceFrom ?? Array.Empty<Game>();
+                var namedgames = freshModList.OtherGames;
                 var validgames = Array.Empty<Game>();
                 foreach (var g in namedgames)
                 {
                     if (canSource.Contains(g) && GameRegistry.Games.ContainsKey(g))
                         validgames.Add(g);
                 }
+
                 var cfg = new InstallerConfiguration
                 {
-                    Game = ModList.GameType,
+                    Game = freshModList.GameType,
                     OtherGames = validgames,
                     Downloads = Installer.DownloadLocation.TargetPath,
                     Install = Installer.Location.TargetPath,
-                    ModList = ModList,
+                    ModList = freshModList,
                     ModlistArchive = WabbajackFileLocation.TargetPath,
                     SystemParameters = _parametersConstructor.Create(),
-                    GameFolder = _gameLocator.GameLocation(ModList.GameType)
+                    GameFolder = _gameLocator.GameLocation(freshModList.GameType)
                 };
+
                 StandardInstaller = StandardInstaller.Create(_serviceProvider, cfg);
 
 

